File: frontend/src/types/safety.ts
import { RiskLevel } from './scholarship';

export interface ThreatFlag {
  id: string;
  category: 'payment_request' | 'otp_pin_request' | 'aadhaar_scraping' | 'domain_spoofing' | 'whatsapp_trap' | 'unrealistic_guarantee' | 'suspicious_contact' | 'urgency_pressure' | 'missing_provider' | 'inconsistent_eligibility';
  severity: 'low' | 'medium' | 'high';
  title: string;
  description: string;
  detectedTextSnippet?: string;
}

export interface SafetyResult {
  scholarshipId: string;
  scholarshipName: string;
  provider: string;
  riskScore: number;
  riskLevel: RiskLevel;
  flags: ThreatFlag[];
  summaryReasoning: string;
  recommendedAction: string;
  checkedAt: string;
  safetyScoreBreakdown: {
    domainAuthority: number;
    feePatternScore: number;
    textAuthenticityScore: number;
    contactVerifiabilityScore: number;
  };
}

File: backend/app/schemas/safety.py
from pydantic import BaseModel
from typing import List, Optional

class ThreatFlagSchema(BaseModel):
    id: str
    category: str
    severity: str
    title: str
    description: str
    detectedTextSnippet: Optional[str] = None

class SafetyScoreBreakdownSchema(BaseModel):
    domainAuthority: int
    feePatternScore: int
    textAuthenticityScore: int
    contactVerifiabilityScore: int

class SafetyResultSchema(BaseModel):
    scholarshipId: str
    scholarshipName: str
    provider: str
    riskScore: int
    riskLevel: str
    flags: List[ThreatFlagSchema]
    summaryReasoning: str
    recommendedAction: str
    checkedAt: str
    safetyScoreBreakdown: SafetyScoreBreakdownSchema
